rust/agama-server/src/network/web.rs

```diff
@@ -237,15 +259,26 @@ async fn connections(
 )]
 async fn add_connection(
     State(state): State<NetworkServiceState>,
-    Json(conn): Json<NetworkConnection>,
+    Json(net_conn): Json<NetworkConnection>,
 ) -> Result<Json<Connection>, NetworkError> {
-    let conn = Connection::try_from(conn)?;
+    let bond = net_conn.clone().bond;
+    let bridge = net_conn.clone().bridge;
+    let conn = Connection::try_from(net_conn)?;
     let id = conn.id.clone();
 
-    state.network.add_connection(conn).await?;
+    state.network.add_connection(conn.clone()).await?;
+
     match state.network.get_connection(&id).await? {
         None => Err(NetworkError::CannotAddConnection(id.clone())),
-        Some(conn) => Ok(Json(conn)),
+        Some(conn) => {
+            if let Some(bond) = bond {
+                state.network.set_ports(conn.clone(), bond.ports).await?;
```

Comment:
Why cloning the connection when you only need the uuid? If you still want to pass the connection, it would be better to just pass a reference.